activation/nipost_test.go

```diff
@@ -1076,27 +1076,30 @@ func TestNIPoSTBuilder_PoETConfigChange(t *testing.T) {
 			})
 			require.NoError(t, err)
 
+			logger := zaptest.NewLogger(t)
 			nb, err := NewNIPostBuilder(
 				db,
 				nil,
-				zaptest.NewLogger(t),
+				logger,
 				PoetConfig{},
 				nil,
 				nil,
 				WithPoetServices(poetProver),
 			)
 			require.NoError(t, err)
 
-			existingRegistrations, err := nb.submitPoetChallenges(
+			defer func() {
+				if r := recover(); r == nil {
+					t.Errorf("Panic is expected")
+				}
+			}()
+
+			nb.submitPoetChallenges(
```

Comment:
You can set a fatal/panic hook in the zap logger to catch it and assert that `logger.Fatal` was called. Example: https://github.com/spacemeshos/go-spacemesh/blob/25222c1771dcf3caeb01e6bb9d31c4ea1d90d450/activation/post_supervisor_test.go#L115-L130

Reply:
zap test logger doesn't call os.Exit(1) by log.Fatal(), so test continues working and calls  poet.Submit(), what is not expected by code logic. So if I add this expected call test logic wouldn't follow normal running code logic, what I thins is incorrect.

Reply:
You could instead return an error and handle it (call log.Fatal()) in the upper layer. This way you could assert that the condition is properly handled while not having to use logger hooks.